docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,347 @@
+                  defaultMessage: 'Select a format',
+                },
+                hidden: true,
+              },
+            },
+            {
+              key: 'hex',
+              value: 'hex',
+              metadatas: {
+                intlLabel: {
+                  id: 'color-picker.color.format.hex',
+                  defaultMessage: 'Hexadecimal',
+                },
+              },
+            },
+            {
+              key: 'rgba',
+              value: 'rgba',
+              metadatas: {
+                intlLabel: {
+                  id: 'color-picker.color.format.rgba',
+                  defaultMessage: 'RGBA',
+                },
+              },
+            },
+          ],
+        },
+      ],
+      advanced: [
+        {
+          sectionTitle: {
+            id: 'global.settings',
+            defaultMessage: 'Settings',
+          },
+          items: [
+            {
+              name: 'required',
+              type: 'checkbox',
+              intlLabel: {
+                id: 'form.attribute.item.requiredField',
+                defaultMessage: 'Required field',
+              },
+              description: {
+                id: 'form.attribute.item.requiredField.description',
+                defaultMessage: "You won't be able to create an entry if this field is empty",
+              },
+            },
+            {
+              name: 'private',
+              type: 'checkbox',
+              intlLabel: {
+                id: 'form.attribute.item.privateField',
+                defaultMessage: 'Private field',
+              },
+              description: {
+                id: 'form.attribute.item.privateField.description',
+                defaultMessage: 'This field will not show up in the API response',
+              },
+            },
+          ],
+        },
+      ],
+      validator: args => ({
+        format: yup.string().required({
+          id: 'options.color-picker.format.error',
+          defaultMessage: 'The color format is required',
+        }),
+      }),
+    },
+  });
+}
+```
+
+:::
+
+::: tip
+The Strapi codebase gives an example of how settings objects can be described: check the [`baseForm.js`](https://github.com/strapi/strapi/blob/21d06ce7d0dc22e6f1096d9afffbb2e2ef65c93c/packages/core/content-type-builder/admin/src/components/FormModal/attributes/baseForm.js) file for the `base` settings and the [`advancedForm.js`](https://github.com/strapi/strapi/blob/21d06ce7d0dc22e6f1096d9afffbb2e2ef65c93c/packages/core/content-type-builder/admin/src/components/FormModal/attributes/advancedForm.js) file for the `advanced` settings. The base form lists the settings items inline but the advanced form gets the items from an [`attributeOptions.js`](https://github.com/strapi/strapi/blob/21d06ce7d0dc22e6f1096d9afffbb2e2ef65c93c/packages/core/content-type-builder/admin/src/components/FormModal/attributes/attributeOptions.js) file.
```

Comment:
I think it might be better to link to the code not as a blob as this will almost certainly never be up to date.

Instead we should swap these blob links out for things like `https://github.com/strapi/strapi/blob/main/packages/core/content-type-builder/admin/src/components/FormModal/attributes/baseForm.js`